wiki/Publishing-guidelines/Writing-functional-specs.md

```diff
@@ -0,0 +1,383 @@
+
+#### Table of contents
+The table of contents in the template is auto generated. As long as you are using proper markdown headings it should find them just fine, and list the page outline up at the top of the page. So make sure to include the table of contents section template in your component wiki page.
+
+#### Dependencies
+The dependency list is one of the most important aspects of the functional specs. It communicates to the developer, and creates almost this written agreement between design and development what components should be used when building a component out. 
+
+###### How to add a dependency
+
+1. Find the dependency list in your component's template.
+2. Add a new list item
+3. Link to the component using the Wiki linking strategy.
+
+<details>
+  <summary><strong>Example</strong></summary><br />
+
+```html
+- [[Component or wiki page name 1]]
+- [[Component or wiki page name 2]]
+- [[Component or wiki page name 3]]
+```
+
+> ![image](https://user-images.githubusercontent.com/3793636/117873919-f6faba80-b265-11eb-81a5-039bdcd822e8.png)  Notice the double brackets with the file name `[[File name]]` in-between. This is different from a markdown link `[text](url)`.
+
+<br />
+</details><br /><br />
+
+#### Used by
+
+By properly linking dependencies we will also run an automation script to collect all the dependencies and create a used by list in addition. Used by components once populated communicates to the reader where this component is used.
+
+###### How to add a used by
+
+> ![image](https://user-images.githubusercontent.com/3793636/117873919-f6faba80-b265-11eb-81a5-039bdcd822e8.png)  Keep dependencies up to date, and leverage used by template section in the component wiki page.
+
+#### Backlinks
+Every time a wiki document references a component other than itself it creates a back link. Those back links are listed out on that component's page as a snippet and categorized by pages. When clicked it will go to the section of the page where it was talked about, and highlight its contents. A backlink doesn't have to be used as a depency to work, and this list is automated. As long as you include it on the page it'll render its content if available.
+
+#### Resources
+The resource section is a numbered list of additional resources not included in the dependency list. For example design specs, box folders, prototypes, and maybe additional documentation are good things you could include here. Follow the numbering standard if possible to star consistent. `r` for resources, and `#` in the list. If for whatever reason we need to remove an item it's best to keep the numbers the name, and skip since there is possible cross page linking and to avoid multiple updates when a resource becomes outdated.
```

Comment:
```suggestion
The resource section is a numbered list of additional resources not included in the dependency list. For example design specs, box folders, prototypes, and any additional documentation that will be helpful. Follow the numbering standard to stay consistent (`r` for resources, and `#` in the list). If you need to remove an item, it's best to keep the numbers the same as there is possible cross page linking. This also avoids multiple updates when a resource becomes outdated.
```

Reply:
Could we write this sentence more like the following? I think it's important to communicate that by "remaining the same" means that the resource itself doesn't change its reference number, and even if we end up skipping a couple numbers it's ok.

> If you need to remove an item, it's best to keep its number the same. Skipping numbers in this scenario is ok, and usually preferred as there is possible cross page linking.

Reply:
Yes, this is good. Two slight tweaks.... for `okay` (always full word) and removed the words `usually` and `possible`. The `could be` handles that it might not be the case.   

> If you need to remove an item, it's best to keep its number the same. Skipping numbers in this scenario is okay, and preferred as there could be cross page linking.